src/core/engine/app_engine_processor.py

```diff
@@ -3,13 +3,15 @@
 from src.core.engine.app_engine_processor_context import AppEngineProcessorContext
 from src.core.engine.app_processor_config import AppProcessorConfig
 from src.core.engine.processors.kitchen_simulator import KitchenSimulator
+from src.core.ioc import get_ioc_instance
 
 
 class AppEngineProcessor:
     def __init__(self):
+        ioc_instance = get_ioc_instance()
         kitchen_simulator_config = AppProcessorConfig("kitchen_simulator_test", 10)
         kitchen_simulator = KitchenSimulator(kitchen_simulator_config)
-        self.app_context = AppEngineProcessorContext(processors=[kitchen_simulator])
+        self.app_context = AppEngineProcessorContext(processors=[kitchen_simulator], ioc_instance=ioc_instance)
```

Comment:
Is not necessary to add the ioc instance since it is a singleton instance for the whole application

Reply:
done, i got the ioc instance in the kitchen simulator, as is the same instance in the whole application